File: tutorial/04-recursive/problems/08-digital-root.ts
export class SoftP0408 {
  digitSum(num: number) {
    let sum = 0;
    while (num > 0) {
      sum += num % 10;
      num = Math.floor(num / 10);
    }
    return sum;
  }
  solution(num: number): number {
    if (num / 10 < 1) return num;
    return this.solution(this.digitSum(num));
  }
}

File: tutorial/04-recursive/problems/08-digital-root.py
class SoftP0408:
    def digitSum(self,num):
        sum=0
        while num>0:
            sum+=num%10
            num=num//10
        return sum
    def solution(self,num):
        if num/10 <1: return num
        return self.solution(self.digitSum(num))